### ISSUE TITLE
[P2] Sigma-gauged CTM crashes when an in-loop chi bump changes the reference shape

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#975]
> 🤖 **AI-generated issue** - written by Codex from the requested iPEPS code review.

## Finding

Combining `forward_gauge="sigma"` with `ctmrg_heuristic_increase_chi=True` crashes as soon as chi increases. The post-bump environment is aligned to a reference that still has the old chi. This fails inside the forward energy evaluation, before any adjoint solve.

Verified at `4090f98be5c3efdce20a28663536f926171f857b`, CPU/JAX x64. Phase gauge with the same bump and sigma gauge without the bump both complete.

## Reproduction

Run with `JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 uv run python`:

```python
import jax.numpy as jnp
import numpy as np
from tenax.algorithms.ipeps import _wrap_as_dense_tensor, heisenberg_gate
from tenax.algorithms._ctm_tensor_convergence import SINGLE_SITE_NEIGHBORS
from tenax.algorithms._ctm_energy_ad import ctm_energy_implicit

A = _wrap_as_dense_tensor(
    jnp.asarray(np.random.default_rng(0).normal(size=(2, 2, 2, 2, 2)))
)
for gauge, bump in [("phase", True), ("sigma", False), ("sigma", True)]:
    try:
        energy = ctm_energy_implicit(
            {(0, 0): A}, SINGLE_SITE_NEIGHBORS, heisenberg_gate(),
            chi=2, max_iter=2, min_iter=2, qr_warmup_steps=0,
            forward_gauge=gauge,
            ctmrg_heuristic_increase_chi=bump,
            ctmrg_heuristic_increase_chi_threshold=1e-12,
            ctmrg_heuristic_increase_chi_step_size=2,
            chi_max=4,
        )
        print(gauge, bump, float(energy))
    except Exception as exc:
        print(gauge, bump, type(exc).__name__, str(exc))
```

Observed:
```text
phase True 0.0920852931315359
sigma False 0.04511200502012644
sigma True TypeError dot_general requires contracting dimensions to have the same shape, got (4,) and (2,).
```

The two-sweep budget intentionally isolates the first dimension change; these control energies are not claims of physical convergence. Directly running `_sigma_gauged_ctm_converge` also confirms that phase+bump returns `final_chi=4` and sigma without bump returns `final_chi=2`.

## Root Cause

[`_run_ctm_loop_with_bump`](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/_ctm_loop_core.py#L208-L227) pads and resweeps at the new chi, then calls `gauge_fix_fn(envs, envs_at_iter_start)`. The second argument still has the original chi.

[`_sigma_gauge_fix_env`](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/_ctm_energy_ad.py#L463-L475) computes `Q_new @ Q_old.conj().T`, which here tries to multiply a `(4, 4)` matrix by a `(2, 2)` matrix.

The existing [bump reference test](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/tests/test_ctm_loop_core.py#L184) uses a no-op gauge spy and asserts old-reference identity. It does not exercise sigma arithmetic. The integration tests in `test_ctm_in_loop_bump_ad_paths.py` select phase gauge.

## Suggested Fix and Coverage

Use a same-chi reference for sigma alignment after growth, or explicitly reset alignment for the dimension-changing transition. Preserve ordinary fixed-chi alignment, and ensure the chosen transition does not undo growth.

Add a real sigma-gauge integration regression that forces chi 2 -> 4, asserts finite output and `final_chi=4`, and exercises a subsequent fixed-chi sweep. Update the spy-only reference test so it does not enforce an incompatible contract.

Related: #492, #514, and #516 introduced/threaded chi growth. This is distinct from #798's corner-axis-order problem: this reproduction uses a single-site unit cell and fails specifically on unequal matrix dimensions.
